src/main/java/org/example/librarymanagement/service/RegistrationService.java

```diff
@@ -0,0 +1,89 @@
+package org.example.librarymanagement.service;
+
+import freemarker.template.TemplateException;
+import jakarta.mail.MessagingException;
+import lombok.AllArgsConstructor;
+import org.example.librarymanagement.dto.UserDTO;
+import org.example.librarymanagement.exception.serviceException.ServiceException;
+import org.example.librarymanagement.model.account.appUser.AppUser;
+import org.example.librarymanagement.model.account.Role;
+import org.example.librarymanagement.common.emailSender.EmailSenderService;
+import org.example.librarymanagement.dto.RegistrationRequestDTO;
+import org.example.librarymanagement.model.token.ConfirmationToken;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.util.ResourceBundle;
+
+@Service
+@AllArgsConstructor
+public class RegistrationService {
```

Comment:
Em tạo interface define method cho service, sau đó handle implement interface đó. Toàn bộ service đều tạo interface